kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java

```diff
@@ -343,12 +472,16 @@ public void start(Map<String, String> properties) {
             config.getBoolean(config.BIGQUERY_PARTITION_DECORATOR_CONFIG);
     if (hasGCSBQTask) {
       startGCSToBQLoadTask();
+    } else if (upsertDelete) {
```

Comment:
I am not sure I am understanding this. It looks like `hasGCSBQTask` is a flag on task0 that is set if `ENABLE_BATCH_CONFIG` is not empty. In which case, a background thread is started to do some batch loading. If both batch loading and upsert delete is enabled, would task0 just get the batch loading background thread and not the merge flush thread?

Reply:
Ah, forgot to add some validation logic for this. Basically, GCS batch loading will be disabled if upsert/delete is enabled and if the user tries to enable both a validation error will be thrown during startup.